packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-javascript-visitor.test.ts.snap

```diff
@@ -17,11 +19,29 @@ export declare class Post {
   readonly comments?: (Comment | null)[] | null;
   readonly createdAt?: string | null;
   readonly updatedAt?: string | null;
-  constructor(init: ModelInit<Post>);
+  
+}
+
+class LazyPost {
+  readonly [__modelMeta__]: {
+    identifier: CompositeIdentifier<Post, ['customPostId', 'title']>;
+    readOnlyFields: 'createdAt' | 'updatedAt';
+  };
+  readonly customPostId: string;
+  readonly title: string;
+  readonly comments: AsyncCollection<Comment>;
+  readonly createdAt?: string | null;
+  readonly updatedAt?: string | null;
+  
+}
+
+export declare type Post = LazyLoading extends Disabled ? EagerPost : LazyPost
+
+export declare const Post: (new (init: ModelInit<Post>) => Post) & Post & {
```

Comment:
slightly confused with this export here. Are we ultimately just exporting a type instead of a class?

Reply:
Previously the the type declaration used a `class` which can be used as a `value` and a `type` in TypeScript. It was not possible to do conditional typing with a class, so we needed to use this technique ([Declaration Merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)).


Without exporting both `type` and `const` here we would get errors:

* without `type`: `'Post' refers to a value, but is being used as a type here. Did you mean 'typeof Post'`
  * We could have customers fix this themselves when upgrading to version 5, but we decided this would be too much friction.
* without `const`: `'Post' only refers to a type, but is being used as a value here.`

Reply:
Got it, so this is needed to support Declaration Merging.